### PR TITLE
feat: Also export function names with snake case

### LINKED ISSUE #430: Export snake case function names
Per https://github.com/near/NEPs/issues/96 meta standard update, all the function names should be snake case. SDKs and APIs on both side should normalize to whatever is the language.

This should allow still to write public export functions in camelCase per JS standard but AS transformer should either replace or better add snake case export functions.

### PR DESCRIPTION
**Pull request by willemneal**
fixes #430

### CODE REVIEW

File: bindgen/src/utils.ts [hunk 1]
```diff
@@ -76,3 +76,9 @@ export function posixRelativePath(from: string, to: string): string {
 export function toString(node: Node): string {
   return ASTBuilder.build(node);
 }
+
+const captialPattern = /([A-Z])/g;
+
+export function make_snake_case(s: string): string {
```
Comment (mehtaphysical) [inline bindgen/src/utils.ts:82]: this should probably be camel case to fit the rest of the functions:

```suggestion
export function makeSnakeCase(s: string): string {
```

File: bindgen/src/JSONBuilder.ts [hunk 2]
```diff
@@ -151,6 +160,17 @@ export class JSONBindingsBuilder extends BaseVisitor {
     super.visit(node);
   }
 
+  make_camel_case_export(
```
Comment (mehtaphysical) [inline bindgen/src/JSONBuilder.ts:163]: this should be camelCase

Comment (mehtaphysical) [reply bindgen/src/JSONBuilder.ts:163]: What do you think about:

```suggestion
  camelCaseToSnakeCaseExport(
```

File: bindgen/src/JSONBuilder.ts [hunk 3]
```diff
@@ -202,7 +222,9 @@ export class JSONBindingsBuilder extends BaseVisitor {
   value_return(val.byteLength, val.dataStart);`);
     }
     this.sb.push(`}
-export { __wrapper_${name} as ${name} }`);
+export { ${WRAPPER_PREFIX + name} as ${name} }
+${this.make_camel_case_export(name)}
```
Comment (mehtaphysical) [inline bindgen/src/JSONBuilder.ts:226]:
```suggestion
${this.camelCaseToSnakeCaseExport(name)}
```

File: bindgen/src/JSONBuilder.ts [hunk 4]
```diff
@@ -151,6 +160,17 @@ export class JSONBindingsBuilder extends BaseVisitor {
     super.visit(node);
   }
 
+  make_camel_case_export(
+    name: string,
+    prefix: string = WRAPPER_PREFIX
+  ): string {
+    let s = make_snake_case(name);
```
Comment (mehtaphysical) [inline bindgen/src/JSONBuilder.ts:167]:
```suggestion
    let s = makeSnakeCase(name);
```

File: bindgen/src/JSONBuilder.ts [hunk 5]
```diff
@@ -12,9 +12,10 @@ import {
   ParameterNode,
 } from "visitor-as/as";
 import { BaseVisitor, utils } from "visitor-as";
-import { SimpleParser, toString } from "./utils";
+import { SimpleParser, toString, make_snake_case } from "./utils";
```
Comment (mehtaphysical) [inline bindgen/src/JSONBuilder.ts:15]:
```suggestion
import { SimpleParser, toString, makeSnakeCase } from "./utils";
```

File: bindgen/src/utils.ts (hunk 1 above)
Comment (willemneal) [reply bindgen/src/utils.ts:82]: Lol fine, I meant it as a joke, but we should maintain a single style.

File: bindgen/src/utils.ts [hunk 6]
```diff
@@ -76,3 +76,9 @@ export function posixRelativePath(from: string, to: string): string {
 export function toString(node: Node): string {
   return ASTBuilder.build(node);
 }
+
+const captialPattern = /([A-Z])/g;
+
+export function makeSnakeCase(s: string): string {
+  return s.replace(captialPattern, "_$1").toLowerCase();
```
Comment (mehtaphysical) [inline bindgen/src/utils.ts:83]: This has a few weird cases, which may be acceptable:

`convertAPIKey` -> `convert_a_p_i_key`
`SomeFunction` -> `_some_function`

Comment (willemneal) [reply bindgen/src/utils.ts:83]: Good point.  I think we to specify that it's not at the beginning. And add `+` to the capital match.  That should solve the issue.  I'll add a test and add your examples as test cases.

**Review by mehtaphysical — APPROVED** [review]
Nice!

**Pull request merged**